docs/proposals/data.md

```diff
@@ -0,0 +1,82 @@
+# Data function support
+
+## Summary
+
+To provide mount data support for envd
+
+
+## Goals
+
+Design a *unified, declarative* interface and underlying architecture to provide dataset in the development environment in a *scalable way*
+
+
+Non-goals: 
+- Support Git-like version control for data
+
+## Common Scenarios
+
+### Possible sources
```

Comment:
Is there any implementation plan for this? Are you aware of any existing solutions that could support multiple sources? This might be helpful as a reference for the range of sources: https://kubernetes.io/docs/concepts/storage/volumes/#volume-types